src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -294,7 +294,12 @@ private async Task UpdateJobRecordAsync(CancellationToken cancellationToken)
                     }
                 }
 
-                await ProcessSearchResultsAsync(searchResult.Results, currentBatchId, anonymizer, cancellationToken);
+                if (_exportJobRecord.ExportType == ExportJobType.All
```

Comment:
nit: would be helpful to add some comments here to explain why we don't process the search results in certain scenarios. Else it might be hard to follow what's happening.